src/sql/workbench/contrib/charts/browser/chartView.ts

```diff
@@ -228,15 +234,33 @@ export class ChartView extends Disposable implements IPanelView {
 		}
 	}
 
-	private shouldGraph() {
+	private showGraph() {
 		// Check if we have the necessary information
 		if (this._currentData && this._queryRunner) {
 			// check if we are being asked to graph something that is available
 			let batch = this._queryRunner.batchSets[this._currentData.batchId];
 			if (batch) {
 				let summary = batch.resultSetSummaries[this._currentData.resultId];
 				if (summary) {
-					this._queryRunner.getQueryRows(0, summary.rowCount, this._currentData.batchId, this._currentData.resultId).then(d => {
+					const maxRowCount = this._configurationService.getValue<IChartsConfiguration>('charts').maxRowCount;
+					const storageKey = 'charts/ignoreMaxRowCountExceededNotification';
+
+					if (summary.rowCount > maxRowCount) {
+						this._adsTelemetryService.sendTelemetryEvent('charts.maxRowCountExceeded');
```

Comment:
Try not to use `sendTelemetryEvent`. You should generally be using the common schema that I defined (see OneNote for details) aka send/create Action/View/Error

So for this along with what I suggested above I would create an action event `showQueryResultsGraph` or something (make sure to put that in TelemetryKeys.ts) and then include the `maxRowCountExceeded` property on it using `withProperties`